### PR TITLE
default_coalib: Rename to system_coafile

### LINKED ISSUE #3389: Change default_coafile naming convention.
Should we change the name of `default_coafile` in `coalib` to `system_coafile`?
It will avoid confusion due to the following lines in `Constants.py`
```
system_coafile = os.path.join(coalib_root, 'default_coafile')

user_coafile = os.path.join(os.path.expanduser('~'), '.coarc')

default_coafile = '.coafile'
```

### PR DESCRIPTION
**Pull request by LeafyLi**
default_coafile: Rename to system_coafile

Closes #3389

### CODE REVIEW

File: coalib/misc/Constants.py [hunk 1]
```diff
@@ -109,11 +109,11 @@
                                     'languages',
                                     'definitions')
 
-system_coafile = os.path.join(coalib_root, 'default_coafile')
+system_coafile = os.path.join(coalib_root, 'system_coafile')
 
 user_coafile = os.path.join(os.path.expanduser('~'), '.coarc')
 
-default_coafile = '.coafile'
+system_coafile = '.coafile'
```
Comment (Mixih) [inline coalib/misc/Constants.py]: Don't change this line... It will break many things

**Comment by LeafyLi** [comment]
how do i fix this?
The requested coafile '/home/ubuntu/.pyenv/versions/3.5.1/lib/python3.5/site-packages/coala-0.10.0.dev99999999999999-py3.5.egg/coalib/system_coafile' does not exist. You can generate it with your current options by adding the `--save` flag or suppress any use of config files with `-I`.

**Comment by yukiisbored** [comment]
@LeafyLi see @Mixih comment

File: coalib/misc/Constants.py (hunk 1 above)
Comment (yukiisbored) [inline coalib/misc/Constants.py]: Revert this line, it causes https://github.com/coala/coala/pull/3397#issuecomment-269145481

Comment (yukiisbored) [inline coalib/misc/Constants.py]: Ah sorry, I didn't see the corresponding issue. See https://github.com/coala/coala/pull/3393#issuecomment-269140598

**Comment by yukiisbored** [comment]
Sorry, I didn't see the corresponding issue. See https://github.com/coala/coala/pull/3393#issuecomment-269140598

**Comment by yukiisbored** [comment]
You also must rename the corresponding file as well

**Comment by sils** [comment]
sorry to spoil the fun, this could break people's systems. If we really wanna rename this we should deprecate it, i.e. if there's a nonempty default_coafile give a warning to the user that that usage is deprecated but use it and we can remove the handling for it next release.

**Comment by sils** [comment]
also some tests seem to be failing, probably need to be adjusted

**Comment by Mixih** [comment]
to fix testing you'll need to rename that setup.py entry. To do what sils said you need to add some logic that checks if there is a pre-exsisting default_coafile and read from there with a  warning

**Comment by LeafyLi** [comment]
So I added some code that checks for a pre-existing default_coafile, but how would I give a warning to the user?

**Comment by sils** [comment]
logging.warning

**Comment by LeafyLi** [comment]
What'd i do wrong this time?

**Comment by sils** [comment]
@Nosferatul please mark PRs WIP if you rebase them

**Comment by sils** [comment]
@LeafyLi please contact us on the chat if you need help

**Comment by sils** [comment]
marking wip because of all the failures etc.

**Comment by Mixih** [comment]
still failing across the board... @LeafyLi  if you expect to need a few iterations of pushing to fix everything, please a `WIP:` to the title of your pull so gitmate knows it's a work in progress.

**Comment by LeafyLi** [comment]
Don't think I'm good enough to contribute.

**Comment by sils** [comment]
Wohooow stop right there

**Comment by sils** [comment]
@LeafyLi we're helping a lot of people getting started and it doesn't matter who's good enough or not. What matters is who *gets* good enough. That's what this newcomer process is about, we're not vetting out people who aren't good enough because if you look at it this way most people aren't. But the whole point of this is to *become good enough*. Every community has it's tricks and things that you have to learn to become good enough to get in. If you do it, you win. Key takeaway is: we all sucked when we started, it's on you to become better and we're here to help you doing that as best as we can.

**Comment by sils** [comment]
you have no idea how much I had to learn when I joined GNOME

**Comment by sils** [comment]
@LeafyLi I've unassignedyou from the issue since you don't seem to want to work on it. Please know that you're very welcome to continuing to work on this or another newcomer issue, just ping us!

**Pull request closed without merging**